components/rule-mgt/org.wso2.carbon.identity.rule.evaluation/src/main/java/org/wso2/carbon/identity/rule/evaluation/core/OperatorRegistry.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.rule.evaluation.core;
+
+import org.wso2.carbon.identity.rule.evaluation.internal.RuleEvaluationComponentServiceHolder;
+import org.wso2.carbon.identity.rule.evaluation.model.Operator;
+import org.wso2.carbon.identity.rule.metadata.service.RuleMetadataService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Registry for operators.
+ * This class is responsible for loading operators from the metadata service,
+ * instantiating its functionality for evaluation.
+ */
+public class OperatorRegistry {
+
+    private static final Map<String, Operator> operators = new HashMap<>();
+
+    private OperatorRegistry() {
+
+    }
+
+    /**
+     * Get operator implementation by name
+     *
+     * @param name Operator name
+     * @return Operator instance
+     */
+    public Operator getOperator(String name) {
+
+        return operators.get(name);
+    }
+
+    /**
+     * Instantiate OperatorRegistry loading operators from metadata service.
+     *
+     * @return OperatorRegistry instance.
+     */
+    public static OperatorRegistry loadOperators() {
+
+        RuleMetadataService ruleMetadataService = RuleEvaluationComponentServiceHolder.getInstance()
+                .getRuleMetadataService();
+
+        ruleMetadataService.getApplicableOperatorsInExpressions()
+                .forEach(operator -> {
+                    switch (operator.getName()) {
+                        case "equals":
```

Comment:
Shall we define class constants for these, as this is used twice

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/6309/commits/3759ec3359d280decdbfa3e007b35654d0d7dccd